### DISCUSSION TITLE
How to restart Swiper js with active Swup?

### DISCUSSION THREAD

**Discussion by AleksandrDomashenko** [nolimits4web/swiper#6507] [Q&A]
There is the following script on the page that changes the content without reloading the page:
```
window._swup = new Swup({
        plugins: [new SwupScrollPlugin()]
 });
```
 On the main page there is a slider with a connected swiper.
 At the initial loading, the slider works, everything is fine.
 But as soon as I go to another page and come back the slider does not work, the buttons are not active...
 
 The sweeper initialization script itself looks like this:

```
 $(document).ready(function(){
    var swiper = new Swiper('.second-frame__slider', {
        slidesPerView: "auto",
        spaceBetween: 30,
        loop: true,
        pagination: {
            el: ".swiper-pagination",
            clickable: true
        },
        navigation: {
            nextEl: ".swiper-button-next",
            prevEl: ".swiper-button-prev"
        },
        keyboard: {
          enabled: true,
          onlyInViewport: false,
        },
    });
});
```
I tried to do it through api methods .UpdateProgress(), .update Size(), .update Slides(), .update().
Nothing happens...

**Comment by lennier83** [comment]
Maybe this helps https://stackoverflow.com/questions/78341659/how-to-restart-swiper-js-with-active-swup
